processors/zipkin/src/test/java/org/hawkular/apm/processor/zipkin/HttpCodesUtilTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.processor.zipkin;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.hawkular.apm.server.api.model.zipkin.BinaryAnnotation;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * @author Pavol Loffay
+ */
+public class HttpCodesUtilTest {
+
+    @Test
+    public void testGetHttpStatusCodes() {
+        BinaryAnnotation http200 = createHttpCodeBinaryAnnotation("200");
+        BinaryAnnotation http400 = createHttpCodeBinaryAnnotation("400");
+        BinaryAnnotation http302 = createHttpCodeBinaryAnnotation("302");
+        BinaryAnnotation http500 = createHttpCodeBinaryAnnotation("500");
+
+        BinaryAnnotation other = new BinaryAnnotation();
+        other.setKey("return code");
+        other.setValue("450");
+
+        List<BinaryAnnotation> binaryAnnotationList = Arrays.asList(http200, http302, http400, http500, other);
+        Assert.assertEquals(4, HttpCodesUtil.getHttpStatusCode(binaryAnnotationList).size());
+
+        binaryAnnotationList = Arrays.asList(other, other);
+        Assert.assertEquals(0, HttpCodesUtil.getHttpStatusCode(binaryAnnotationList).size());
+    }
+
+    @Test
+    public void testGetHttpStatusCodesOfNullAndEmptyInput() {
+        Assert.assertTrue(HttpCodesUtil.getHttpStatusCode(null).isEmpty());
+        Assert.assertTrue(HttpCodesUtil.getHttpStatusCode(Collections.emptyList()).isEmpty());
+    }
+
+    @Test
+    public void testGetErrors() {
+        HttpCodesUtil.HttpCode code200 = new HttpCodesUtil.HttpCode(200, "Ok");
+        HttpCodesUtil.HttpCode code302 = new HttpCodesUtil.HttpCode(302, "Found");
+        HttpCodesUtil.HttpCode code400 = new HttpCodesUtil.HttpCode(400, "Bad Request");
+        HttpCodesUtil.HttpCode code501 = new HttpCodesUtil.HttpCode(501, "Not Implemented");
+
+        List<HttpCodesUtil.HttpCode> codeList = Arrays.asList(code200, code400, code400, code302, code501);
+
+        List<HttpCodesUtil.HttpCode> clientOrServerErrors = HttpCodesUtil.getClientOrServerErrors(codeList);
+        Assert.assertEquals(3, clientOrServerErrors.size());
+        Assert.assertEquals(Arrays.asList(code400, code400, code501), clientOrServerErrors);
```

Comment:
Is this assertion "order dependent"? If I change the order on `codeList`, would the test break? And more importantly: *should* the test break?